coloc/coloc_runner.py

```diff
@@ -0,0 +1,251 @@
+    eqtl_r = eqtl_r %>% select(beta, varbeta, position, snp, N, MAF)
+
+    eqtl_r = eqtl_r %>% as.list()
+    eqtl_r$type = 'quant'
+
+    my.res <- coloc.abf(dataset1=gwas_r,
+                    dataset2=eqtl_r)
+
+    p_df <- data.frame(gene,my.res$summary[1], my.res$summary[2], my.res$summary[3], my.res$summary[4], my.res$summary[5], my.res$summary[6])
+    names(p_df) <- c('gene', 'nsnps_coloc_tested','PP.H0.abf','PP.H1.abf','PP.H2.abf','PP.H3.abf','PP.H4.abf')
+    ''',
+    )
+
+    # convert to pandas df
+    with (ro.default_converter + pandas2ri.converter).context():
+        pd_p4_df = ro.conversion.get_conversion().rpy2py(ro.r('p_df'))
+
+    # add cell type and chrom annotation to df
+    pd_p4_df['celltype'] = celltype
+    pd_p4_df['chrom'] = eqtl['chr'].iloc[0]
+
+    # write to GCS
+    pd_p4_df.to_csv(
+        coloc_results_file,
+        sep='\t',
+        index=False,
+    )
+
+
+@click.option(
+    '--egenes-files-path',
+    help='Path to the gene-level summary files',
+    default='gs://cpg-bioheart-test-analysis/saige-qtl/bioheart_n990_and_tob_n1055/241004_n100/output_files/summary_stats',
+)
+@click.option(
+    '--snp-cis-dir',
+    help='Path to the directory containing the SNP cis results',
+    default='gs://cpg-bioheart-test-analysis/saige-qtl/bioheart_n990_and_tob_n1055/241004_n100/output_files',
+)
+@click.option(
+    '--snp-gwas-file',
+    help='Path to the SNP GWAS file',
+    default='gs://cpg-bioheart-test/str/gwas_catalog/gcst/gcst-gwas-catalogs/GCST011071_parsed.tsv',
+)
+@click.option(
+    '--celltypes', help='Cell types to run, single str, comma separated', default='ASDC'
+)
+@click.option('--cis-window-size', help='Cis window size used', default=100000)
+@click.option('--fdr-threshold', help='FDR threshold', default=0.05)
+@click.option(
+    '--max-parallel-jobs', help='Maximum number of parallel jobs to run', default=500
+)
+@click.option(
+    '--pheno-output-name', help='Phenotype output name', default='covid_GCST011071'
+)
+@click.option('--job-cpu', help='Number of CPUs to use for each job', default=0.25)
+@click.command()
+def main(
+    snp_cis_dir: str,
+    egenes_files_path: str,
+    celltypes: str,
+    snp_gwas_file: str,
+    pheno_output_name: str,
+    max_parallel_jobs: int,
+    cis_window_size: int,
+    fdr_threshold: float,
+    job_cpu: float,
+):
+    # Setup MAX concurrency by genes
+    _dependent_jobs: list[hb.batch.job.Job] = []
+
+    def manage_concurrency_for_job(job: hb.batch.job.Job):
+        """
+        To avoid having too many jobs running at once, we have to limit concurrency.
+        """
+        if len(_dependent_jobs) >= max_parallel_jobs:
+            job.depends_on(_dependent_jobs[-max_parallel_jobs])
+        _dependent_jobs.append(job)
+
+    # read in gene annotation file
+    var_table = pd.read_csv(
+        'gs://cpg-bioheart-test/saige-qtl/300-libraries/combined_anndata_obs_vars/300_libraries_concatenated_harmony_filtered_vars.csv',
```

Comment:
This file is quite confusing, but if I understand correctly you're only using it to set the coordinates for the windows and it's ok that it's the same for every celltype etc? Ideally this should probably get swapped out with a parameter at some point.

Reply:
This is just the `adata.var` object: the sc expression file comes in [AnnData format](https://anndata.readthedocs.io/en/stable/) with the count matrix as the main info, and `obs` and `var` tables including information on cells and genes (only) respectively.

Here I have just extracted the `var` object the info in which mainly comes from Gencode.

But yes to making this a parameter, I'll do this now

Reply:
https://github.com/populationgenomics/saige-tenk10k/pull/214/commits/b0b93332e01e7749c47e9c63abcab863d9942bb7